src/Sulu/Component/Webspace/Url/Replacer.php

```diff
@@ -89,17 +90,46 @@ public function replaceSegment($url, $segment)
     /**
      * {@inheritdoc}
      */
+    public function hasHostReplacer($url)
+    {
+        return $this->hasReplacer($url, self::REPLACER_HOST);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function replaceHost($url, $host)
+    {
+        return $this->replace($url, self::REPLACER_HOST, $host);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
     public function replace($url, $replacer, $value)
     {
         return str_replace($replacer, $value, $url);
     }
 
+    public function replaceAll($url, $replacers)
```

Comment:
I would have put that logic into the `replace` function, since it doesn't make too much sense for me to have a public function simply wrapping `str_replace`... But if it is a too big BC break I could live with it, since it was already like that before.